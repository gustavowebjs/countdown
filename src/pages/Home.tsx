import { ReactElement, useState } from 'react';
import { Button, CountDownTimer, Input } from '../components';
import { TimeFormat } from '../components/CountDownTimer/types';
import * as S from './styles';

function Home(): ReactElement {
  const [activeTimer, setActiveTimer] = useState(false);
  const [timerSpeed, setTimerSpeed] = useState(1000);
  const [timerFormat, setTimerFormat] = useState<TimeFormat>({
    minutes: 0,
    seconds: 0,
  });
  const [initialMinutes, setInitialMinutes] = useState<number>();

  const handleInputChange = (
    event: React.ChangeEvent<HTMLInputElement>
  ): void => {
    const { value, name } = event.target;

    if (name === 'minutes') {
      setInitialMinutes(Number(value));
    }
  };

  const handleStartTimer = (): void => {
    if (initialMinutes) {
      setTimerFormat({
        minutes: initialMinutes,
        seconds: 0,
      });

      setActiveTimer(true);
    }
  };

  return (
    <S.Container>
      <S.CountDown>
        <S.WrapperTimer>
          <S.WrapperInput>
            <Input
              name="minutes"
              onChange={handleInputChange}
              label="Set minutes"
              type="number"
              placeholder="MM"
              length={2}
            />
            <Button
              className="outline"
              onClick={() => handleStartTimer()}
              style={{ marginLeft: 5 }}
            >
              Start
            </Button>
          </S.WrapperInput>

          <S.Timer>
            <CountDownTimer
              speed={timerSpeed}
              active={activeTimer}
              timeFormat={timerFormat}
            />
          </S.Timer>

          <S.SpeedButtons>
            <Button
              className={`outline ${timerSpeed === 1000 ? 'active' : ''}`}
              size="small"
              onClick={() => setTimerSpeed(1000)}
            >
              Normal
            </Button>
            <Button
              className={`outline ${timerSpeed === 750 ? 'active' : ''}`}
              size="small"
              onClick={() => setTimerSpeed(750)}
            >
              1.5x
            </Button>
            <Button
              className={`outline ${timerSpeed === 500 ? 'active' : ''}`}
              size="small"
              onClick={() => setTimerSpeed(500)}
            >
              2.0x
            </Button>
          </S.SpeedButtons>

          <Button onClick={() => setActiveTimer(!activeTimer)}>
            {activeTimer ? 'Pause' : 'Start'}
          </Button>
        </S.WrapperTimer>
      </S.CountDown>
    </S.Container>
  );
}

export default Home;
