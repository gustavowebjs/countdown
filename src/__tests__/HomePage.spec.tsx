import { shallow } from 'enzyme';
import Home from '../pages/Home';

describe('Home Page', () => {
  it('Should Home Page render correctly', () => {
    const wrapper = shallow(<Home />);

    const startButton = wrapper.find('Button');
    expect(startButton.length).toEqual(1);

    const countdown = wrapper.find('CountDownTimer');
    expect(countdown.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
