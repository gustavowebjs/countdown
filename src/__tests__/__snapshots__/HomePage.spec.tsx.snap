// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should Home Page render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <ForwardRef(Input)
          label="Set minutes"
          length={2}
          name="minutes"
          onChange={[Function]}
          placeholder="MM"
          type="number"
        />
        <Button
          className="outline"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          <ForwardRef(SvgStart) />
          Start
        </Button>
      </styled.div>
      <styled.div>
        <CountDownTimer
          active={false}
          onComplete={[Function]}
          speed={1000}
          timeFormat={
            Object {
              "minutes": 0,
              "seconds": 0,
            }
          }
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
